add all Rucio lfns2pfns args. to getPFN. Fix #10056

belforte opened the pull request:
Fixes #10056 

#### Status
ready

#### Description
introduce new optional parameters for Rucio.client.getPFN
to be able to use all functionalities of Rucio python lnfs2pfns API as per
https://rucio.readthedocs.io/en/latest/api/rse.html

#### Is it backward compatible (if not, which system it affects?)
YES

#### Related PRs
#10121 on CRAB branch 1.3.6.crab

#### External dependencies / deployment changes
None

belforte committed:
add all Rucio lfns2pfns args. to getPFN. Fix #10056

amaltaro commented on src/python/WMCore/Services/Rucio/Rucio.py:
```diff
@@ -290,16 +290,19 @@ def getReplicaInfoForBlocks(self, **kwargs):
         return result
 
-    def getPFN(self, site, lfns, protocol=None):
+    def getPFN(self, site, lfns, protocol=None, protocol_domain='ALL', operation=None):
```
Stefano, can you please adopt the lower camel case variable name? So changing `protocol_domain` to `protocolDomain`?
I've been trying to avoid this never ending mix of variable naming conventions in WMCore.

belforte replied:
You are right. There's no need to keep same name as RUCIO API in our code. Changed.

amaltaro reviewed (commented):
Changes look good to me, but I'd like to have that variable name updated if possible.
Stefano, once you make that change, please squash your two commits.

amaltaro commented:
@belforte thanks Stefano. Could you please squash those 2 commits into a single one? In case you need some instructions on how to accomplish that, I usually refer to this short and clear doc: https://steveklabnik.com/writing/how-to-squash-commits-in-a-github-pull-request

belforte commented:
rebased and squashed. Yeah. I have added tips from that URL to may cheat sheet and had used already to squash my few commits while testing, so I did not realize it was needed again. Sorry.

amaltaro reviewed (approved):
Thanks, Stefano!

amaltaro commented:
Just spotted badly encoded chars in this PR (though a git diff)
```
Supported are <E2><80><98>LAN<E2><80><99>, <E2><80><98>WAN<E2><80><99>, and <E2><80><98>ALL<E2><80><99> (as default)
```

will get that fixed in another PR.
